src/fallback.ts

```diff
@@ -14,35 +14,30 @@
  * limitations under the License.
  */
 
-// Not all browsers support `TextEncoder`. The following `require` will
-// provide a fast UTF8-only replacement for those browsers that don't support
-// text encoding natively.
-import {isBrowser} from './isbrowser';
-let needTextEncoderPolyfill = false;
-
-if (
-  isBrowser() &&
-  // eslint-disable-next-line node/no-unsupported-features/node-builtins
-  (typeof TextEncoder === 'undefined' || typeof TextDecoder === 'undefined')
-) {
-  needTextEncoderPolyfill = true;
-}
-if (
-  typeof process !== 'undefined' &&
-  process?.versions?.node &&
-  process?.versions?.node.match(/^10\./)
-) {
-  // Node.js 10 does not have global TextDecoder
-  // TODO(@alexander-fenster): remove this logic after Node.js 10 is EOL.
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  const util = require('util');
-  Object.assign(global, {
-    TextDecoder: util.TextDecoder,
-    TextEncoder: util.TextEncoder,
-  });
-}
-if (needTextEncoderPolyfill) {
-  require('fast-text-encoding');
+/* global window */
+/* global AbortController */
+
+import {
+  hasWindowFetch,
+  hasTextEncoder,
+  hasTextDecoder,
+  isNodeJS,
+  hasAbortController,
+} from './featureDetection';
+
+if (!hasTextEncoder() || !hasTextDecoder()) {
+  if (isNodeJS()) {
```

Comment:
I generally don't like the idea of trying to figure out if something is Node.js or not.  The philosophy behind feature detection is to say which browser/env it is doesn't matter, and instead focus on the availability of the API itself.  Things like electron and webpack make figuring out "is this node?" a fools errand.  All that having been said - I don't have a ready made suggestion on how to do this better :)

Reply:
I thought about this. There are several use cases of `isNodeJS()`, what I _actually_ need is `hasUtil()` (can `require("util")` and `hasEmitWarning()`. Checking the latter is straightforward (and it might make sense to have a separate "feature" for `process.emitWarning`), but I don't know a good way to check if `require('util').TextEncoder` and `require('util').TextDecoder` are available - so I'm just assuming here that if it's Node.js, it must be able to do `require("util")`.  Do you think there might be a better way?

(and yes, we'll drop Node v10 and this particular `if` on line 29 will disappear)